server/src/main/java/org/eclipse/lsp/cobol/core/engine/CobolLanguageEngine.java

```diff
@@ -126,15 +126,21 @@ public CobolLanguageEngine(
             positionMapping,
             messageService,
             subroutineService);
-    Node rootNode = visitor.visit(tree).get(0);
-    SyntaxTreeEngine syntaxTreeEngine = new SyntaxTreeEngine(rootNode);
-    accumulatedErrors.addAll(syntaxTreeEngine.processTree());
+    List<Node> syntaxTree = visitor.visit(tree);
+    if (syntaxTree.size() == 1) {
+      Node rootNode = syntaxTree.get(0);
+      SyntaxTreeEngine syntaxTreeEngine = new SyntaxTreeEngine(rootNode);
+      accumulatedErrors.addAll(syntaxTreeEngine.processTree());
+    } else
+      LOG.warn("The root node for syntax tree was not constructed");
```

Comment:
It will be logged even for the empty files, that is a valid case. I would recommend using a less severe level here

Reply:
Ok, makes sense